packages/flutter_migrate/lib/src/custom_merge.dart

```diff
@@ -82,20 +81,20 @@ class MetadataCustomMerge extends CustomMerge {
   MigrateConfig mergeMigrateConfig(
       MigrateConfig current, MigrateConfig target) {
     // Create the superset of current and target platforms with baseRevision updated to be that of target.
-    final Map<SupportedPlatform, MigratePlatformConfig> platformConfigs =
-        <SupportedPlatform, MigratePlatformConfig>{};
-    for (final MapEntry<SupportedPlatform, MigratePlatformConfig> entry
+    final Map<FlutterProjectComponent, MigratePlatformConfig> platformConfigs =
+        <FlutterProjectComponent, MigratePlatformConfig>{};
```

Comment:
Should the value be a `Platform`-related thing if the key is no longer platform-specific? Does that type need a name change or split as well?

Reply:
Yeah, it does. But I think I'll leave a TODO here and do it in a separate PR to sync it here and in the flutter_tools repo. Id prefer to keep these concepts synced.